### DISCUSSION TITLE
cache action has existential crisis

### DISCUSSION THREAD

**Discussion by incertia** [community/community#25030] [Actions]
<p>I am trying to setup caching for a build pipeline in a piece of software, but the cache action never seems to play nicely.</p>
<p>The cache pre-run looks like</p>
<pre><code>Run actions/cache@v1
  with:
    path: .github-cache
    key: Linux-libs
Cache not found for input keys: Linux-libs.
</code></pre>
<p>while the cache post-run looks like</p>
<pre><code>Post job cleanup.
[warning]Cache already exists. Scope: refs/heads/github-ci, Key: Linux-libs, Version: (null)
</code></pre>
<p>The relevant sections of the workflow file are</p>
<pre><code>- name: Cache Libraries
        id: cache-libs
        uses: actions/cache@v1
        with:
          path: .github-cache
          key: ${{ runner.os }}-libs

      - name: Build Libraries
        run: |
          .github/scripts/get-cache.sh
          .github/scripts/install-libsecp256k1.sh
          .github/scripts/install-libff.sh
          .github/scripts/put-cache.sh
        env:
          HOST_OS: ${{ runner.os }}
</code></pre>
<p>tl;dr the scripts try to get the libraries from the cache directory and install them to $HOME/.local/lib and the installation scripts only run if the library cannot be found. We then put copies into the cache directory so the post-job can run correctly.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>This is an issue about cache action, please refer to this issue : <a href="https://github.com/actions/cache/issues/144" rel="nofollow noopener">https://github.com/actions/cache/issues/144</a></p>
<p>As a workaround, please try to change your key. Changing your key will allow you to create a new cache. </p>
